lms/services/canvas_api.py

```diff
@@ -68,10 +68,13 @@ def list_files(self, course_id):
 
         :rtype: list(dict)
         """
-        return self._helper.validated_response(
-            self._helper.list_files_request(self._oauth2_token.access_token, course_id),
+        oauth2_token = self._oauth2_token
+
+        return self.send_with_refresh_and_retry(
+            self._helper.list_files_request(oauth2_token.access_token, course_id),
             CanvasListFilesResponseSchema,
-        ).parsed_params
+            oauth2_token.refresh_token,
+        )
```

Comment:
This is just the exact same request and response, but now sent via the new `send_with_refresh_and_retry()`